### DISCUSSION TITLE
Is there a way to enable normal mode for input only for certain actions?

### DISCUSSION THREAD

**Discussion by pewpewnor** [nvim-neo-tree/neo-tree.nvim#1379] [Q&A] [answered]
Hi everyone,

I understand that setting enable_normal_mode_for_inputs = true would apply to every file actions like creating and renaming a file. I'm just wondering, is there a way for me to configure neo-tree to only enable this normal mode behavior specifically for renaming and not for other actions such as creating a new file?

Perhaps there is a way to remap r to a callback function to do this?

Thanks.

**Comment by pysan3** [comment, marked as answer]
Update to the latest main branch, and you'll have this feature.

https://github.com/nvim-neo-tree/neo-tree.nvim/pull/1372

However, the argument given to this handler is [`NuiInput`](https://github.com/MunifTanjim/nui.nvim/blob/main/lua/nui/input/README.md) object and to my knowledge it is not possible to detect which type of prompt it is.

You do have `input.bufnr` and `input.winid` so it is not impossible to fetch eg the title or buffer content (_Enter new name for_, ...).

TLDR; no, and it is very difficult to implement due to how the code is structured.
